docs/4-functional-programming/homework/src/main.ts

```diff
@@ -1,20 +1,54 @@
-import { Either, fromPromise, ap, right, getOrElse, flatten } from './fp/either';
+import { Either, fromPromise, ap, right, getOrElse, flatten, isRight, left } from './fp/either';
 import { pipe } from './fp/utils';
+import { fromNullable } from './fp/maybe';
+import { ordNumber } from './fp/ord';
 import { fetchClient, fetchExecutor } from './fetching';
 import { ClientUser, ExecutorUser } from './types';
 
 type Response<R> = Promise<Either<string, R>>
 
 const getExecutor = (): Response<ExecutorUser> => fromPromise(fetchExecutor());
-const getClients = (): Response<Array<ClientUser>> => fromPromise(fetchClient());
+const getClients = (): Response<Array<ClientUser>> => {
+  return fromPromise(fetchClient()).then((users) => {
+    return isRight(users) ? {...users, right: users.right.map(item => ({...item, demands: fromNullable(item.demands)}))} : users;
+  }).catch((err) => err);
+};
 
 export enum SortBy {
   distance = 'distance',
   reward = 'reward',
 }
 
+interface Position {
+  x: number,
+  y: number,
+}
+
+const calculateDistance = (executorPosition: Position, clientPosition: Position) => {
+  return Math.sqrt(Math.pow(clientPosition.x - executorPosition.x, 2) + Math.pow(clientPosition.y - executorPosition.y, 2)).toFixed(3);
+}
+const sortByParameter = (clients: Array<ClientUser>, executor: ExecutorUser, parameter: SortBy) => {
+  if (parameter === SortBy.reward) return clients.sort((x, y) => ordNumber.compare(x.reward, y.reward)).reverse();
```

Comment:
1. it is better not to write everything in one line
2. best practice is to have early return, it means: 
`if (true) {
    return;
  }
return logic